src/OpenColorIO/Mutex.h

```diff
@@ -24,18 +25,26 @@ class DebugLock
     DebugLock() = default;
     DebugLock(const DebugLock &) = delete;
     DebugLock& operator=(const DebugLock &) = delete;
-    ~DebugLock() { assert(!m_locked); }
-
-    void lock()     { assert(!m_locked); m_mutex.lock();   m_locked = true;  }
-    void unlock()   { assert(m_locked);  m_mutex.unlock(); m_locked = false; }
-    void try_lock() { assert(!m_locked); m_locked = m_mutex.try_lock(); }
-
-    bool locked() const { return m_locked; }
+    ~DebugLock() { assert(m_owner == std::thread::id()); }
+
+    void lock()
+    {
+        assert(m_owner != std::this_thread::get_id());
+        m_mutex.lock();
+        m_owner = std::this_thread::get_id();
+    }
+    void unlock()
+    {
+        assert(m_owner == std::this_thread::get_id());
```

Comment:
The following line seems useless.

Reply:
It checks that the same thread is holding the lock before unlocking. It seems useful to me, but I might be missing something?

Reply:
I refer to line 39 (and not line 38 which is fine).
In normal use case, `m_owner` variable already owns the right thread id.

Reply:
Ah, that's to indicate there is no owner. `std::thread::id()` sounds like it might return the current thread id, but it's just the default constructor with an empty thread id.

Reply:
Now, I understand (Refer to https://en.cppreference.com/w/cpp/thread/thread/id)